### PR TITLE
adding support for nd sum in parametrized expressions

### PR DESCRIPTION
**Pull request by Transurgeon**
## Description
Please include a short summary of the change.
This PR adapts the SciPy backend sum function to support parametrized sums. 
I introduced a new helper function ``get_sum_coeff_matrix`` since I noticed that it was being reused for both cases (non-dpp and dpp). 
This PR keeps the old functionality the same and adds a general case for when ``axis != None``.
Issue link (if applicable):

## Type of change
- [x] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/tests/test_expressions.py [hunk 1]
```diff
@@ -1571,6 +1571,16 @@ def test_nd_sum(self, axis) -> None:
         prob.solve(canon_backend=cp.SCIPY_CANON_BACKEND)
         assert np.allclose(expr.value, y)
 
+    @pytest.mark.parametrize("axis", [(0),(1),(2),((0,1)),((0,2)),((2,1))])
+    def test_nd_parametrized_sum(self, axis) -> None:
+        param = cp.Parameter((2,2,2))
+        param.value = np.ones((2,2,2))
```
Comment (SteveDiamond) [inline cvxpy/tests/test_expressions.py:1577]: can you do it with the value np.reshape(np.arange(8), (2,2,2))? using all ones might mask bugs.

Comment (Transurgeon) [reply cvxpy/tests/test_expressions.py:1577]: ok let me change that.

Comment (Transurgeon) [reply cvxpy/tests/test_expressions.py]: should be resolved once all the CI passes.

**Review by SteveDiamond — APPROVED** [review]
Nice job!

**Pull request merged**